### ISSUE TITLE
"Leave charger on after charging cycle" switch

### ISSUE THREAD

**Issue by tubalainen** [jonasbkarlsson/ev_smart_charging#65]
**Is your feature request related to a problem? Please describe.**
No, well ish.

**Describe the solution you'd like**
Hej Jonas! After the charging window has been finalized and the car has reached its target SOC, could we please implement a `switch` to leave the charger on? (And reset the cycle when the charger cable is unplugged, via an automation for example).
This new switch feature would have to interact with the new `switch.ev_smart_charging_continuous_charging_preferred`. The cannot co-exist (or be enabled at the same time).

The reason for this is during Swedish winter (du vet vad jag menar) it is by the manufacturer recommended to keep the cable "plugged in and enabled" when the car is idling. And also to draw power from the grid to perform "scheduled departures" (preheating etc), the preheating and warmup normally occurs after the charging cycle slot.

**Describe alternatives you've considered**
Doing an automation when the  `sensor.ev_smart_charging_charging` goes to `off` and then re-enable it again. And reset the logic after the cable has been unplugged. The problem with this one is that this might cause the car to through a tantrum and say "problems with charger" when it is drawing some current when the charger "switch" is toggled like this.
I´d rather have the option to keep it on without having an automation to toggle it.

**Additional context**

**Comment by jonasbkarlsson** [comment]
Thanks for the suggestion. I do recognize the need to enable the charger before departure in order to do preheating during winter. :-)

However, your suggestion is to leave the charger on if the car has reached its target SOC. This would mean that the charger would not be left on if the target SOC has not been reached (e.g. if the price was too high compared with the `Electricity price limit`. Which would mean that the car sometimes will not be preheated, which wouldn't be that good...

So basically, this would only be meaningful for those that do not use the `Electricity price limit`. Is that the case for you?

**Comment by tubalainen** [comment]
> Thanks for the suggestion. I do recognize the need to enable the charger before departure in order to do preheating during winter. :-)
> 
> However, your suggestion is to leave the charger on if the car has reached its target SOC. This would mean that the charger would not be left on if the target SOC has not been reached (e.g. if the price was too high compared with the `Electricity price limit`. Which would mean that the car sometimes will not be preheated, which wouldn't be that good...
> 
> So basically, this would only be meaningful for those that do not use the `Electricity price limit`. Is that the case for you?

Correct. I do not use the `Electricity price limit` feature. Hence I want to be able to drive my car - in all cases. If the prices are like insane I will do more drastic measures to my power consumption (such as switch of the house completely). And possibly go to a supercharger nearby instead and juice up for 8 kr/kW (fingers crossed).

So, if implemented - my suggested `switch` will only apply if you havnt set `Electricity price limit` and that triggered.
It could possibly apply to 'switch.ev_smart_charging_continuous_charging_preferred' as well (if the logic is based on weather the SOC has been reached).

The electricity cost of having the charger "on" after the charging cycle is completed is a of great benefits in both cases (with out without the continuous charging enabled). Well, if you do not use the 'Electricity price limit' feature that is.

**Comment by tubalainen** [comment]
And - If the car is still plugged in and did reach the SOC the charger will just continue to be "on" - until the cable is unplugged and then the cycle and logic will start over.

**Comment by jonasbkarlsson** [comment]
That sound like a good solution. So if `switch.ev_smart_charging_apply_price_limit` is off or `Electricity price limit` is set to 0, then this new `switch` will keep the charger on after the last scheduled charging. This should be ok independent of the setting of the `switch.ev_smart_charging_continuous_charging_preferred`.

And then it will be the user's responsibiity to turn off the charger. The suggested way to do that is to turn off `switch.ev_smart_charging_ev_connected` when the car is unplugged.

**Comment by jonasbkarlsson** [comment]
Have done a pre-release [v1.4.0.dev1](https://github.com/jonasbkarlsson/ev_smart_charging/releases/tag/v1.4.0.dev1) that includes a new switch `switch.ev_smart_charging_keep_charger_on`. Have not updated the documentation yet.

**Comment by tubalainen** [comment]
Tusen tack! Works great.

I will make a PR for the documentation.

**Comment by jonasbkarlsson** [comment]
Now included in [v1.4.0](https://github.com/jonasbkarlsson/ev_smart_charging/releases/tag/v1.4.0)!

After the updates I did today, I don't see an absolute need to use the `switch.ev_smart_charging_ev_connected` to turn off the charger with respect to this feature. So I removed that note in the documentation. But of course, there are still other reasons to use that switch, such as making sure the charger is turned off while the EV is disconnected.
